src/Tools/CodeGenerator/Plugins/OnnxRuntimePlugin.py

```diff
@@ -487,74 +516,107 @@ def CreateMacro(
             )
 
         else:
-            # We need to create one function per output type
+            # We need to create one combined function for all output types
+            all_output_types = []
+            all_input_types = []
             for output_type, input_types in six.iteritems(output_type_mappings):
-                type_constraints = OrderedDict()
-
-                if len(input_types) == 1:
-                    input_type = "tensor({})".format(input_types[0])
+                all_output_types.append(re.sub("_t$", "", output_type))
+                all_input_types.extend(re.sub("_t$","", value) for value in input_types)
+
+            type_constraints = OrderedDict()
+
+            if len(all_input_types) == 1:
+                input_type = "tensor({})".format(all_input_types[0])
+            else:
+                input_type = "InputT"
+                type_constraints[input_type] = [
+                    "{}".format(value.replace("std::", ""))
+                    for value in all_input_types
+                ]
+
+            if custom_struct_data is None or all_output_types[0] not in custom_struct_data:
+                if len(all_output_types) == 1:
+                    output_type = all_output_types[0]
+                    output_statements = [
+                        CreateOutputStatement(
+                            "tensor({})".format(output_type.replace("std::", "")),
+                            item.output_description,
+                        ),
+                    ]
                 else:
-                    input_type = "InputT"
-                    type_constraints[input_type] = [
-                        "tensor({})".format(value.replace("std::", ""))
-                        for value in input_types
+                    output_type = "OutputT"
+                    type_constraints[output_type] = [
+                        "{}".format(value.replace("std::", ""))
+                        for value in all_output_types
                     ]
-
-                if custom_struct_data is None or output_type not in custom_struct_data:
                     output_statements = [
                         CreateOutputStatement(
-                            "tensor({})".format(output_type.replace("std::", "")),
+                            "OutputT",
                             item.output_description,
                         ),
                     ]
 
+                if (len(all_output_types) == 1):

```

Comment:
Does it make sense to put the suffix creation code into the if/else statement above?

Reply:
Yeah. Initially there was only one suffix so it was below. I didn't think about moving it up when I added in the second one. I'll move it up.